security/proto/istioca.proto

```diff
@@ -0,0 +1,42 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package istio.v1.auth;
+
+// Certificate request message.
+message IstioCertificateRequest {
+  // PEM-encoded certificate request.
+  string csr = 1;
+  // Optional subject ID field.
+  string subject_id = 2;
+  // Optional: requested certificate validity period, in seconds.
+  int64 validity_duration = 3;
+}
+
+// Certificate response message.
+message IstioCertificateResponse {
+  // PEM-encoded certificate chain.
+  // Leaf cert is element '0'. Root cert is element 'n'.
+  repeated string cert_chain = 1;
+}
+
+// Service for managing certificates issued by the CA.
+service IstioCertificateService {
+  // Using provided CSR, returns a signed certificate.
+  rpc CreateCertificate(IstioCertificateRequest)
+      returns (IstioCertificateResponse) {
+  }
```

Comment:
Just so I understand the model, node agents ie daemon sets will request csrs from citadel and then the node agent will sign pod certificates and serve pod certificates to sidecar Envoy over SDS? 

Won’t this put the CA in the critical path for all pod creation events

Reply:
Actually in OSS (k8s), the Envoy requests cert from node agent via SDS. Node agent creates the key pair and sends the CSR to Citadel with pod JWT. Citadel signs the cert and returns it to node agent, and node agent sends the cert to Envoy through SDS.

Reply:
Sure.

We're aware of the plan to remove envoy sidecar for control plane components. And their mTLS/security requirement definitely will be covered before we turn this on.

Reply:
@rshriram We talked with @qiwzhang who is working on SDS support for Envoy. Actually if a SDS server can't response with the valid certificates, only that particular listener will be affected. Envoy in general will not crash and other listener not using the SDS can also work.

Reply:
Thanks for the reply @incfly